### PR TITLE
[i18n] Language switcher for anon users

### PR DESCRIPTION
**Pull request by rajatvijay**
Related #2927 

**Technical details**

This PR does the following things -

1. Language switcher for the anon pages.
2. Save the selected language in the cookies to be preserved during re-loads. 

Notice: 
1. No matter what the user selects here, the `preferred_language` saved in the DB always takes the preference to translated pages once the user has logged in. 
2. The default value of `preferred_language` user's model attribute is `en` so that attribute will always have a value. The ideal way would have been to give cookies lang preference for the users who are logging in for the first time. But this needs some discussion. 
3. This logic can be found [here](https://github.com/centerofci/mathesar/pull/3103/files#diff-62a976429d7be143bb6f8ffd9e3cef6d3f139936eefc227c867495d2cdeee631R22) but the language switch functionality for anon users is introduced here hence mentioned this product related point in this PR

**Screenshots**

Screenshot 2023-07-27 at 4 04 09 PM

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [ ] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by rajatvijay** [comment]
@ghislaineguerin @kgodey assigned you for a product related review.

**Review by silentninja — APPROVED** [review]
Looks good!

File: mathesar/templates/mathesar/login_base.html
Comment (silentninja) [inline mathesar/templates/mathesar/login_base.html:202]: Can you add comments where the `set_language` url coming from

**Comment by ghislaineguerin** [comment]
@rajatvijay should the language switcher have some kind of label? Something like 'Default Language' or 'UI Language'

**Review by kgodey — APPROVED** [review]
Looks fine to me.

File: mathesar/templates/mathesar/login_base.html
Comment (seancolsen) [inline mathesar/templates/mathesar/login_base.html:204]: Can you add `aria-label="Display Language"` to the `<select>` element?

**Review by seancolsen — CHANGES_REQUESTED** [review]
This looks mostly good. I've requested one minor change. Also, we'll need a way to hide this UI if we merge this before the Japanese translation is complete.

**Comment by ghislaineguerin** [comment]
Looks good to me.

File: mathesar/templates/mathesar/login_base.html
Comment (rajatvijay) [reply mathesar/templates/mathesar/login_base.html:155]: de03c76

**Comment by rajatvijay** [comment]
Commented out the UI element to hide it for untile complete i18n work is merged. de03c76

**Review by seancolsen — APPROVED** [review]
Looks good now 👍🏻

**Comment by seancolsen** [comment]
I'm holding off merge since this is stacked on top of #3103. @rajatvijay feel free to merge this yourself once that PR is merged.

**Pull request merged**
